ui/src/app/cohort-search/modifier-page/modifier-page.component.tsx

```diff
@@ -361,15 +356,20 @@ export const ModifierPage = fp.flow(withCurrentWorkspace(), withCurrentCohortSea
               break;
             case ModifierType.EVENTDATE:
               const formatted = values.map(val => moment(val, 'YYYY-MM-DD', true).isValid()
-                    ? moment(val).format('YYYY-MM-DD') : undefined);
+                ? moment(val).format('YYYY-MM-DD') : undefined);
               acc.push({name, operator, operands: formatted.filter(val => !!val)});
               break;
             default:
-              acc.push({name, operator, operands: values.filter(val => !!val)});
+              acc.push({name, operator, operands: values.filter(val => !['', null, undefined].includes(val))});
```

Comment:
Also update the filter condition here, since `0` values in the number inputs were getting filtered out, resulting in a bad request for the calculate api call.